pkg/controller/add.go

```diff
@@ -81,8 +82,10 @@ func AddToManagerWithOptions(ctx context.Context, mgr manager.Manager, opts *Add
 func infrastructurePredicate() predicate.TypedFuncs[*extensionsv1alpha1.Infrastructure] {
 	return predicate.TypedFuncs[*extensionsv1alpha1.Infrastructure]{
 		UpdateFunc: func(e event.TypedUpdateEvent[*extensionsv1alpha1.Infrastructure]) bool {
-			// We want to reconcile if the status of the Infrastructure changed
-			return !apiequality.Semantic.DeepEqual(e.ObjectOld.Status, e.ObjectNew.Status)
+			// We want to reconcile if the EgressCIDRs of the Infrastructure changed
+			slices.Sort(e.ObjectOld.Status.EgressCIDRs)
```

Comment:
Predicates must not mutate objects, because doing so also modifies the object in the watch cache (informer). controller-runtime passes pointers to the cache objects to the predicates, not deep copies.

You have two options:
- copy the slices before sorting
- ignore the order and assume it rarely changes and the consequences are negligible

Reply:
ah yes I've read about that somewhere. made a copy and removed the deepequal as we don't need it for comparing just slices.
should be good2go now ;)